logictest: add multiregion configuration

otan opened the pull request:
Add a logictest configuration that has three regions, with three
nodes in three different AZs.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55784)
<!-- Reviewable:end -->

otan commented:
can't seem to add @cockroachdb/multiregion as reviewers, but here's a CC.

ajstorm reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55784)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @otan)

___
*[pkg/sql/logictest/logic.go, line 642 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55784#-MKB9XSZD0f0gbbfH-uu:-MKB9XSZD0f0gbbfH-uv:b-hvalns) ([raw file](https://github.com/cockroachdb/cockroach/blob/facba9ddb99a4c788e7b5800a68be13b2c975a57/pkg/sql/logictest/logic.go#L642)):*
> ```Go
> 	{
> 		name:              "multiregion",
> 		numNodes:          9,
> ```

Seems like it could be error prone to have the map below depend on this size.  Is there some way we can generate regions (and az's) below to match whatever number is up here?  I'm worried that someone will change this number, say to 15, and we'll have a bunch of nodes without localities set.
___
*[pkg/sql/logictest/logic.go, line 647 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55784#-MKB78se7bZ5d9Ec6m5I:-MKB78se7bZ5d9Ec6m5J:b-nqvfx1) ([raw file](https://github.com/cockroachdb/cockroach/blob/facba9ddb99a4c788e7b5800a68be13b2c975a57/pkg/sql/logictest/logic.go#L647)):*
> ```Go
> 			1: {
> 				Tiers: []roachpb.Tier{
> 					{Key: "region", Value: "test"},
> ```

I'd think a more realistic setup would be to have 3 nodes in each region, with three different availability zones.  Also, I'm thinking it might be best to have the regions be "-" delimited and the az's be either numeric or chars to match what AWS, GCP and Azure do.


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55784)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm)

___
*[pkg/sql/logictest/logic.go, line 642 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55784#-MKB9XSZD0f0gbbfH-uu:-MKBOlyO8ncrcOLqZyrx:bs29j0s) ([raw file](https://github.com/cockroachdb/cockroach/blob/facba9ddb99a4c788e7b5800a68be13b2c975a57/pkg/sql/logictest/logic.go#L642)):*
<details><summary><i>Previously, ajstorm wrote…</i></summary><blockquote>

Seems like it could be error prone to have the map below depend on this size.  Is there some way we can generate regions (and az's) below to match whatever number is up here?  I'm worried that someone will change this number, say to 15, and we'll have a bunch of nodes without localities set.
</blockquote></details>

i can deliberately rename to `multiregion-9node` to avoid this issue. i can also throw an assert.
___
*[pkg/sql/logictest/logic.go, line 647 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55784#-MKB78se7bZ5d9Ec6m5I:-MKBOuaxA9M1q37mu8yp:b-v5visi) ([raw file](https://github.com/cockroachdb/cockroach/blob/facba9ddb99a4c788e7b5800a68be13b2c975a57/pkg/sql/logictest/logic.go#L647)):*
<details><summary><i>Previously, ajstorm wrote…</i></summary><blockquote>

I'd think a more realistic setup would be to have 3 nodes in each region, with three different availability zones.  Also, I'm thinking it might be best to have the regions be "-" delimited and the az's be either numeric or chars to match what AWS, GCP and Azure do.
</blockquote></details>

works for me. the spec had the space.


<!-- Sent from Reviewable.io -->

ajstorm reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55784)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm)

___
*[pkg/sql/logictest/logic.go, line 642 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55784#-MKB9XSZD0f0gbbfH-uu:-MKBjY6i7w1cb9vKAiY7:b-f63vaa) ([raw file](https://github.com/cockroachdb/cockroach/blob/facba9ddb99a4c788e7b5800a68be13b2c975a57/pkg/sql/logictest/logic.go#L642)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

i can deliberately rename to `multiregion-9node` to avoid this issue. i can also throw an assert.
</blockquote></details>

Both of those sound good.


<!-- Sent from Reviewable.io -->

otan committed:
logictest: add multiregion configuration

Add a logictest configuration that has three regions, with three
nodes in three different AZs.

Release note: None

ajstorm reviewed (commented):
> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55784)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm)



<!-- Sent from Reviewable.io -->

ajstorm reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55784)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

ajstorm commented:
LGTM

otan commented:
thanks!

bors r=ajstorm
